drivers/disk/ramdisk.c

```diff
@@ -38,6 +39,14 @@ static int disk_ram_access_init(struct disk_info *disk)
 static int disk_ram_access_read(struct disk_info *disk, uint8_t *buff,
 				uint32_t sector, uint32_t count)
 {
+	uint32_t last_sector = sector + count;
+
+	if (last_sector < sector || last_sector > RAMDISK_SECTOR_COUNT) {
+		LOG_ERR("Sector %u is outside the range %u",
```

Comment:
I think that instead of `u` the `PRIu32` should be used. The `u` formatting is int side dependent while types here are hardwired to 32bit size.